packages/ember-data/lib/system/model/model.js

```diff
@@ -781,7 +782,11 @@ var Model = Ember.Object.extend(Ember.Evented, {
 
   hasMany: function() {
     Ember.assert("The `hasMany` method is not available on DS.Model, a DS.Snapshot was probably expected. Are you passing a DS.Model instead of a DS.Snapshot to your serializer?", false);
-  }
+  },
+
+  setId: Ember.observer('id', function () {
```

Comment:
@stefanpenner Trying out the code you provided I'm seeing `model.id` return the correct value however, `model.get('id')` appears to return a cached `undefined` value when the id property is watched by the template. I thought `notifyPropertyChange` would invalidate the cache...